src/views/catalog/templatescatalog/components/TemplatesCatalogDrawer/TemplatesCatalogDrawerCreateForm.tsx

```diff
@@ -51,37 +51,60 @@ import {
 } from '@patternfly/react-core';
 
 import AuthorizedSSHKey from './AuthorizedSSHKey';
+import { useDrawerContext } from './useDrawerContext';
+import {
+  doNameParameterExists,
+  getTemplateNameParameterValue,
+  setTemplateParameterValue,
+  uploadFiles,
+} from './utils';
 
 type TemplatesCatalogDrawerCreateFormProps = {
   authorizedSSHKey: string;
   canQuickCreate: boolean;
-  initialVMName?: string;
-  isBootSourceAvailable: boolean;
   namespace: string;
   onCancel: () => void;
   subscriptionData: RHELAutomaticSubscriptionData;
-  template: V1Template;
 };
 
 export const TemplatesCatalogDrawerCreateForm: FC<TemplatesCatalogDrawerCreateFormProps> = memo(
-  ({
-    authorizedSSHKey,
-    canQuickCreate,
-    initialVMName,
-    isBootSourceAvailable,
-    namespace,
-    onCancel,
-    subscriptionData,
-    template,
-  }) => {
+  ({ authorizedSSHKey, canQuickCreate, namespace, onCancel, subscriptionData }) => {
     const history = useHistory();
     const { t } = useKubevirtTranslation();
-    const { updateTabsData, updateVM, vm } = useWizardVMContext();
 
-    const [vmName, setVMName] = useState(initialVMName || '');
+    const { updateTabsData, updateVM } = useWizardVMContext();
+
+    const {
+      cdFile,
+      diskFile,
+      isBootSourceAvailable,
+      setTemplate,
+      template,
+      uploadCDData,
+      uploadDiskData,
+      vm,
+    } = useDrawerContext();
+
+    const nameParameterExists = doNameParameterExists(template);
+
+    const [vmName, setVMName] = useState(generateVMName(template));
+
+    const nameField = nameParameterExists
+      ? getTemplateNameParameterValue(template) || vmName
+      : vmName;
+
+    const onVMNameChange = (newName: string) => {
+      nameParameterExists
+        ? setTemplate((draftTemplate) =>
+            setTemplateParameterValue(draftTemplate, NAME_INPUT_FIELD, newName),
+          )
+        : setVMName(vmName);
```

Comment:
if setTemplateParameterValue return a value, please change the name of the function as its. very confusing, to use this name inside another setter